### DISCUSSION TITLE
Create / update Elasticsearch ETL operations via node.js client?

### DISCUSSION THREAD

**Discussion by nickchampion** [ravendb/ravendb#15334] [General]
Hi,

I've looked through the node client code and can see we can use this to create a SQL ETL task

```
const operation = new AddEtlOperation(etlConfiguration);
const etlResult = await store.maintenance.send(operation);
```

But is it possible to create an Elasticsearch ETL operation via the client, I want to be able to keep our transform scripts in source control and deploy updates via our deployment pipeline?

Also is it possible to load documents in a transform script and is it advisable if it is?

Thanks

**Comment by ayende** [comment]
The node.js client hasn't been updated yet to support elastic search ETL.
You can use the C# / JVM client API for that. 
Or add external support for that, basically, you'll need to port this:

https://github.com/ravendb/ravendb-jvm-client/tree/v5.4/src/main/java/net/ravendb/client/documents/operations/etl/elasticSearch
